test/Locker.test.ts

```diff
@@ -0,0 +1,56 @@
+import { Contract } from "ethers";
+import { expect } from "./chai-setup";
+import { pack, keccak256 } from "@ethersproject/solidity";
+import { ethers } from "hardhat";
+import {
+  setupNftGemGovernor
+} from './fixtures/Governance.fixture';
+
+describe('Locker Test Suite', function () {
+  let locker: Contract;
+  let NFTGemMultiToken: any;
+  const tokenHash = keccak256(['bytes'], [pack(['string'], ['Test Token'])]);
+  beforeEach(async () => {
+    // Deploy Locker Contract on each test iteration
+    locker = await (await ethers.getContractFactory('Locker')).deploy();
+    const setupNftGemGovernorResult = await setupNftGemGovernor();
+    NFTGemMultiToken = setupNftGemGovernorResult.NFTGemMultiToken;
+    const [sender] = await ethers.getSigners();
+    // Mint token to sender to make sure the sender have enough balance
+    await NFTGemMultiToken.mint(sender.address, tokenHash, 100);
+    await NFTGemMultiToken.connect(sender).setApprovalForAll(locker.address, true);
```

Comment:
what's the connect for? do it like this: await NFTGemMultiToken.setApprovalForAll(locker.address, true);